### PR TITLE
Corrects #2613 - Fix lost update loop

### PR DESCRIPTION
**Pull request by kolloch**

### CODE REVIEW

File: src/test/scala/mesosphere/marathon/core/task/tracker/impl/TaskStatusUpdateProcessorImplTest.scala [hunk 1]
```diff
@@ -9,118 +9,160 @@ import mesosphere.marathon.Protos.MarathonTask
...
-import org.mockito.{ Matchers, ArgumentCaptor, Mockito }
+import org.mockito.ArgumentCaptor
+import org.scalatest.GivenWhenThen
+import org.scalatest.concurrent.ScalaFutures
 
 import scala.concurrent.duration._
 import scala.concurrent.{ Await, Future }
 
-class TaskStatusUpdateProcessorImplTest extends MarathonSpec {
+class TaskStatusUpdateProcessorImplTest extends MarathonSpec with Mockito with ScalaFutures with GivenWhenThen {
```
Comment (meichstedt) [inline src/test/scala/mesosphere/marathon/core/task/tracker/impl/TaskStatusUpdateProcessorImplTest.scala]: :+1: for using ScalaFutures – there are still `Await.result` calls in the test, can you please replace them with `.futureValue`?

File: src/test/scala/mesosphere/marathon/core/task/tracker/impl/TaskStatusUpdateProcessorImplTest.scala [hunk 2]
```diff
@@ -9,118 +9,160 @@ import mesosphere.marathon.Protos.MarathonTask
...
+    verify(f.launchQueue).addDelay(app)
+    And("the launch queue has been notified")
+    verify(f.launchQueue).notifyOfTaskUpdate(any)
+    And("the update has been acknowledged")
+    verify(f.schedulerDriver).acknowledgeStatusUpdate(status)
+
+    And("the appropriate event got published on the event stream")
+    val eventCaptor = ArgumentCaptor.forClass(classOf[MesosStatusUpdateEvent])
+    verify(f.eventBus).publish(eventCaptor.capture())
+    assert(eventCaptor.getValue != null)
```
Comment (meichstedt) [inline src/test/scala/mesosphere/marathon/core/task/tracker/impl/TaskStatusUpdateProcessorImplTest.scala]: Do you mind using Matchers here? They're used more often that asserts in our tests, so I'd consider them conventional.

**Comment by meichstedt** [comment]
@kolloch Two comments and a failing test (possibly unrelated, but tests used to be stable in the last weeks iirc) – can you please address?

File: src/test/scala/mesosphere/marathon/core/task/tracker/impl/TaskStatusUpdateProcessorImplTest.scala (hunk 1 above)
Comment (kolloch) [inline src/test/scala/mesosphere/marathon/core/task/tracker/impl/TaskStatusUpdateProcessorImplTest.scala]: ack

File: src/test/scala/mesosphere/marathon/core/task/tracker/impl/TaskStatusUpdateProcessorImplTest.scala (hunk 2 above)
Comment (kolloch) [inline src/test/scala/mesosphere/marathon/core/task/tracker/impl/TaskStatusUpdateProcessorImplTest.scala]: ack

**Comment by kolloch** [comment]
Thanks for your comments, @meichstedt !

I hopefully addressed them now.

**Comment by meichstedt** [comment]
:+1: :ship:

**Pull request merged**